### PR TITLE
Refs #23919 -- Removed misc Python 2/3 references.

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

File: django/db/migrations/serializer.py [hunk 1]
```diff
@@ -155,11 +155,6 @@ def serialize(self):
             raise ValueError("Cannot serialize function: lambda")
         if self.value.__module__ is None:
             raise ValueError("Cannot serialize function %r: No module" % self.value)
-        # Python 3 is a lot easier, and only uses this branch if it's not local.
```
Comment (timgraham) [inline django/db/migrations/serializer.py]: @andrewgodwin, can you confirm this removal? I'm not sure if some test coverage might be missing considering the comment implies Python 3 does use this branch.

Comment (andrewgodwin) [inline django/db/migrations/serializer.py]: Python 3 does indeed use this branch as `__qualname__` is a 3.3 and up feature, I believe. This code triggers for an object that is in a global scope (so it can give a full import path), and the code below it triggers for Python 2 and objects in a local scope. If anything could be removed, it's some of the code below, but I'm not sure which combination at the moment, as some of it is definitely used.

In short, though, don't remove this!

Comment (andrewgodwin) [inline django/db/migrations/serializer.py]: (and yes, it could do with some more test coverage so we can figure that out)

Comment (timgraham) [inline django/db/migrations/serializer.py]: In the second commit (along with some other cleanups), I readded a test similar to the one removed in https://github.com/django/django/commit/c716fe87821df00f9f03ecc761c914d1682591a2#diff-d073e708462e9844f28fdfa568043d51 which it seems should have always run on Python 3 (without the expectation of an exception).
